web/web.go

```diff
@@ -0,0 +1,694 @@
+// 7) Add rateConversion() in schema conversion, ddl and report APIs.
+// 8) Add an overview in summary report API
+var mysqlTypeMap = make(map[string][]typeIssue)
+var postgresTypeMap = make(map[string][]typeIssue)
+
+func homeLink(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "Welcome to Harbourbridge!")
+	w.WriteHeader(http.StatusOK)
+}
+
+// Driver config is used for direct database connection.
+type DriverConfig struct {
+	Driver   string `json:"Driver"`
+	Host     string `json:"Host"`
+	Port     string `json:"Port"`
+	Database string `json:"Database"`
+	User     string `json:"User"`
+	Password string `json:"Password"`
+}
+
+func databaseConnection(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var config DriverConfig
+	err = json.Unmarshal(reqBody, &config)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	var dataSourceName string
+	switch config.Driver {
+	case "postgres":
+		dataSourceName = fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable", config.Host, config.Port, config.User, config.Password, config.Database)
+	case "mysql":
+		dataSourceName = fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", config.User, config.Password, config.Host, config.Port, config.Database)
+	default:
+		http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", config.Driver), http.StatusBadRequest)
+		return
+	}
+	sourceDB, err := sql.Open(config.Driver, dataSourceName)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("SQL connection error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	// Open doesn't open a connection. Validate database connection.
+	err = sourceDB.Ping()
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Connection Error: %v. Check Configuration again.", err), http.StatusInternalServerError)
+		return
+	}
+	app.sourceDB = sourceDB
+	app.dbName = config.Database
+	app.driver = config.Driver
+	w.WriteHeader(http.StatusOK)
+}
+
+func convertSchemaSQL(w http.ResponseWriter, r *http.Request) {
+	if app.sourceDB == nil || app.dbName == "" || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Database is not configured or Database connection is lost. Please set configuration and connect to database."), http.StatusNotFound)
+		return
+	}
+	conv := internal.MakeConv()
+	var err error
+	switch app.driver {
+	case "mysql":
+		err = mysql.ProcessInfoSchema(conv, app.sourceDB, app.dbName)
+	case "postgres":
+		err = postgres.ProcessInfoSchema(conv, app.sourceDB)
+	default:
+		http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), http.StatusBadRequest)
+		return
+	}
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Schema Conversion Error : %v", err), http.StatusNotFound)
+		return
+	}
+	app.conv = nil
```

Comment:
Spurious line?
Its is immediately overwritten by the next line.

Reply:
yeah, this is not needed, as we are completely replacing app.conv here